acceptance/indy_acceptance/indy-tests/test_scripts/acceptance_tests/special_case_trust_anchor_role.py

```diff
@@ -0,0 +1,168 @@
+"""
+Created on Nov 8, 2017
+
+@author: khoi.ngo
+
+Containing test scripts of test scenario 11: special case for TrustAnchor role.
+"""
+
+# !/usr/bin/env python3.6
+import json
+from indy import ledger, signus
+from indy.error import IndyError
+from libraries.constant import Constant, Colors, Roles
+from libraries.common import Common
+from libraries.utils import perform, generate_random_string, exit_if_exception, perform_with_expected_code
+from test_scripts.test_scenario_base import TestScenarioBase
+
+
+# -----------------------------------------------------------------------------------------
+# This will run acceptance tests that will validate the add/remove roles functionality.
+# -----------------------------------------------------------------------------------------
+
+
+class TestScenario11(TestScenarioBase):
+
+    async def execute_test_steps(self):
+        print("Test Scenario 11 -> started")
+        # Declare all values use in the test
+        seed_trustee1 = generate_random_string(prefix="Trustee1", size=32)
+        seed_trustee2 = generate_random_string(prefix="Trustee2", size=32)
+        seed_steward1 = generate_random_string(prefix="Steward1", size=32)
+        seed_steward2 = generate_random_string(prefix="Steward2", size=32)
+        seed_trustanchor1 = generate_random_string(prefix="TrustAnchor1", size=32)
+        seed_trustanchor2 = generate_random_string(prefix="TrustAnchor2", size=32)
+        seed_trustanchor3 = generate_random_string(prefix="TrustAnchor3", size=32)
+        try:
+            # 1. Create ledger config from genesis txn file  ---------------------------------------------------------
+            self.steps.add_step("Create and open pool Ledger")
+            returned_code = await perform(self.steps, Common.prepare_pool_and_wallet, self.pool_name,
+                                          self.wallet_name, self.pool_genesis_txn_file)
+
+            self.pool_handle, self.wallet_handle = exit_if_exception(returned_code)
+
+            # 2. Create DIDs ----------------------------------------------------
+            self.steps.add_step("Create DIDs")
+            returned_code = await perform(self.steps, signus.create_and_store_my_did,
+                                          self.wallet_handle, json.dumps({"seed": Constant.seed_default_trustee}))
+            default_trustee_did = returned_code[0] if len(returned_code) == 2 else (None, None)
+
+            returned_code = await perform(self.steps, signus.create_and_store_my_did,
```

Comment:
The status of the current step is overwritten many times. So FAILED status may be overwritten by PASSED that is obviously not the expected behavior.

Reply:
This issue has not been fixed.

Reply:
Please refer to https://github.com/hyperledger/indy-node/pull/456/commits/84876afddf1eb493652ca089e8449e9325c0a3eb#diff-c811b8ce6def5210a2c02403e1237026
then expand the line 47 for your review.

Reply:
Please see my recent comment to the similar finding above.

Reply:
Please refer to latest code for detail update.

Reply:
Just noticed the freeze logic in `Step.set_status`. It has fixed the issue.